src/nbl/disable_3rdparty_warnings.h

```diff
@@ -0,0 +1,13 @@
+#ifndef _NBL_DISABLE_3RDPARTY_WARNINGS_
+#define _NBL_DISABLE_3RDPARTY_WARNINGS_
+
+#ifdef _MSC_VER
+
+#define NBL_PUSH_DISABLE_WARNINGS __pragma(warning(push, 0))
+#define NBL_POP_DISABLE_WARNINGS __pragma(warning(pop))
+
+#else
+
```

Comment:
you need to handle it because you will end up with compile errors if `_MSC_VER` is not defined, add

```c++
#define NBL_PUSH_DISABLE_WARNINGS
#define NBL_POP_DISABLE_WARNINGS
```

to make it empty